dlp/api/Snippets/InspectStorageWithSampling.cs

```diff
@@ -0,0 +1,144 @@
+        var dlp = DlpServiceClient.Create();
+
+        // Construct storage config.
+        var storageConfig = new StorageConfig
+        {
+            CloudStorageOptions = new CloudStorageOptions
+            {
+                FileSet = new CloudStorageOptions.Types.FileSet
+                {
+                    Url = GcsFileUri
+                },
+                BytesLimitPerFile = 200,
+                FileTypes = { new FileType[] { FileType.Csv } },
+                FilesLimitPercent = 90,
+                SampleMethod = CloudStorageOptions.Types.SampleMethod.RandomStart
+            }
+        };
+
+        // Construct the inspect config.
+        var inspectConfig = new InspectConfig
+        {
+            InfoTypes =
+            {
+                infoTypes ?? new InfoType[] { new InfoType { Name = "PERSON_NAME" } }
+            },
+            IncludeQuote = true,
+            MinLikelihood = minLikelihood
+        };
+
+        // Construct the pubsub action.
+        var actions = new Google.Cloud.Dlp.V2.Action[]
+        {
+            new Google.Cloud.Dlp.V2.Action
+            {
+                PubSub = new Google.Cloud.Dlp.V2.Action.Types.PublishToPubSub
+                {
+                    Topic = $"projects/{projectId}/topics/{topicId}"
+                }
+            }
+        };
+
+        // Construct the inspect job config using actions.
+        var inspectJob = new InspectJobConfig
+        {
+            StorageConfig = storageConfig,
+            InspectConfig = inspectConfig,
+            Actions = { actions }
+        };
+
+        // Issue Create Dlp Job Request
+        var client = DlpServiceClient.Create();
+        var request = new CreateDlpJobRequest
+        {
+            InspectJob = inspectJob,
+            ParentAsLocationName = new LocationName(projectId, "global"),
+        };
+
+        // We need created job name
+        var dlpJob = dlp.CreateDlpJob(request);
+        var jobName = dlpJob.Name;
+
+        // Listen to pub/sub for the job
+        var subscriptionName = new SubscriptionName(projectId, subId);
+        var subscriber = SubscriberClient.CreateAsync(
+            subscriptionName).Result;
+
+        // SimpleSubscriber runs your message handle function on multiple threads to maximize throughput.
+        var done = new ManualResetEventSlim(false);
+        subscriber.StartAsync((PubsubMessage message, CancellationToken cancel) =>
+        {
+            if (message.Attributes["DlpJobName"] == jobName)
+            {
+                // Wait for DLP API results to become consistent
+                Thread.Sleep(500);
+                done.Set();
+                return Task.FromResult(SubscriberClient.Reply.Ack);
+            }
+            else
+            {
+                return Task.FromResult(SubscriberClient.Reply.Nack);
+            }
+        });
+
+        // 3 minute timeout; may not work for large jobs
+        done.Wait(TimeSpan.FromMinutes(3));
```

Comment:
I am concerned the test can be flaky if the job takes longer than 3 min. The Java sample seems to wait for 15 min.

Reply:
See this example to find out how you can stop the subscriber after the correct message has been received without having to explicitly wait.

https://cloud.google.com/dotnet/docs/reference/Google.Cloud.PubSub.V1/latest#sample-code

Reply:
![image](https://user-images.githubusercontent.com/77317274/234556889-2df36870-81de-44d2-adf9-43b42075846e.png)

Like this?

Reply:
Nope, similar to the code you had before, which I guess shows how to handle this DLP messages. Something like this (But it is best if you push the code so I can make suggestions on it instead of a screenshot, thanks)

```
            if (message.Attributes["DlpJobName"] == jobName)
            {
                // Wait for DLP API results to become consistent
                subscriberStop = subscriber.StopAsync(TimeSpan.FromMiliseconds(500))
                Task.FromResult(SubscriberClient.Reply.Ack);
            }
            else
            {
                return Task.FromResult(SubscriberClient.Reply.Nack);
            }

.......
// This is outside the message processing delegate.
await subscriberStop;
```

Reply:
I have pushed the code as per your suggestion but the code is getting terminated right after `StartAsync` method. I tried to debug the code but couldn't figure out what the actual issue is.

Reply:
Sure thanks. I tried awaiting `StartAsync` method but still getting same behaviour.

Reply:
I did not have time to look today, and I'm not sure I'll have time to look until next week.

Reply:
I have updated the code which will not explicitly wait. It will stop the subscriber right after receiving the correct message. But yes we will have to wait for the `StartAsync` task to get completed before moving further. Let me know if this works. Thanks.